src/source/Ice/IceAgent.c

```diff
@@ -1392,22 +1392,28 @@ STATUS iceAgentInitRelayCandidate(PIceAgent pIceAgent)
     UINT32 j, i;
     PKvsIpAddress pSocketAddrForTurn = NULL;
     BOOL locked = FALSE;
+    CHAR ipAddrStr[KVS_IP_ADDRESS_STRING_BUFFER_LEN];
+    PTurnConnectionTracker pCurrentTurnConnectionTracker = NULL;
 
     CHK(pIceAgent != NULL, STATUS_NULL_ARG);
-
     for (j = 0; j < pIceAgent->iceServersCount; j++) {
         if (pIceAgent->iceServers[j].isTurn) {
-
+            pSocketAddrForTurn = NULL;
             // if an VPN interface (isPointToPoint is TRUE) is found, use that instead
             for(i = 0; i < pIceAgent->localNetworkInterfaceCount; ++i) {
                 if(pIceAgent->localNetworkInterfaces[i].family == pIceAgent->iceServers[j].ipAddress.family &&
-                        (pSocketAddrForTurn == NULL || pIceAgent->localNetworkInterfaces[i].isPointToPoint)) {
+                   (pSocketAddrForTurn == NULL || pIceAgent->localNetworkInterfaces[i].isPointToPoint)) {
                     pSocketAddrForTurn = &pIceAgent->localNetworkInterfaces[i];
                 }
             }
 
-            CHK(pSocketAddrForTurn != NULL, STATUS_TURN_CONNECTION_NO_HOST_INTERFACE_FOUND);
+            if (pSocketAddrForTurn == NULL) {
+                getIpAddrStr(&pIceAgent->iceServers[j].ipAddress, ipAddrStr, SIZEOF(ipAddrStr));
+                DLOGW("No suitable local interface found for turn server %s", ipAddrStr);
+                continue;
```

Comment:
Please don't use continue; statement. Use else clause here. I understand it's adding an extra layer of indentation. See if you can split/outsource the content into a separate fn.